src/lib/components/TokenSelect/TokenButton.tsx

```diff
@@ -42,10 +39,37 @@ interface TokenButtonProps {
 export default function TokenButton({ value, collapsed, disabled, onClick }: TokenButtonProps) {
   const buttonBackgroundColor = useMemo(() => (value ? 'interactive' : 'accent'), [value])
   const contentColor = useMemo(() => (value || disabled ? 'onInteractive' : 'onAccent'), [value, disabled])
+
+  // Transition the button if transitioning from a disabled state. This makes initialization cleaner.
+  const button = useRef<HTMLButtonElement>(null)
+  const [row, setRow] = useState<HTMLDivElement | null>(null)
+  const empty = !value
+  const [shouldTransition, setShouldTransition] = useState(true)
+  useEffect(() => {
+    if (disabled) {
+      setShouldTransition(true)
+    } else if (empty) {
+      setShouldTransition(false)
+    }
+  }, [disabled, empty])
+  const style = useMemo(() => {
+    // Widths may only be transitioned using absolute values, so we must compute it.
+    const width = row?.clientWidth
+    return { width: shouldTransition && width ? width + 10 : undefined }
+  }, [row, shouldTransition])
+
   return (
-    <StyledTokenButton onClick={onClick} empty={!value} color={buttonBackgroundColor} disabled={disabled}>
+    <StyledTokenButton
+      onClick={onClick}
+      color={buttonBackgroundColor}
+      disabled={disabled}
+      bleedIn={shouldTransition}
+      onAnimationEnd={() => setShouldTransition(false)}
```

Comment:
Alternatively, `Button` could be wrapped so that it exposes its own `onBleedIn` handler (at which point they could be renamed to something like `animateEnable`/`onAnimateEnable`), but that seemed like overkill for this one-off.